### PR TITLE
Fixed hindi number to english number word in number recognition of eg…

### PR DESCRIPTION
**Pull request by SanchitTaliyan**
…gquiz

### CODE REVIEW

**Comment by SanchitTaliyan** [comment]
|PR checklist|Status| 
|-----|:------:| 
|Is issue created in GitHub?[No] 
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes| 
|Is code analysis task executed and issues resolved in the CPP/native code?|Yes| 
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA| 
|Is code commenting done properly for the changes made?|Yes| 
|Is correct variable naming convention followed?|Yes| 
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes| 
|Are the message strings used in the application taken from strings.xml file?|No| 
|Are the permissions which are not required removed from AndroidManifest.java file?|NA| 
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA| 
|Are the generated files correctly configured in .gitignore?|Yes| 
|Are the release signing configurations removed from the gradle files?|NA| 
|Is commented code removed from all files?|Yes| 
|Are updated images compressed (< 100 KB)?|NA| 
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 1]
```diff
@@ -179,15 +210,14 @@ void EggQuizScene::createSingleDigitNumbersView() {
...
-    
     auto questionBox = Scale9Sprite::create(partsPath+"pretest-questionbox.png");
     questionBox->setPreferredSize(Size(2360,800));
     questionBox->setAnchorPoint(Vec2::ANCHOR_MIDDLE_BOTTOM);
     questionBox->setPosition(_gameNode->getContentSize().width/2,680);
     _gameNode->addChild(questionBox);
     
     auto speaker = Speaker::create(SpeakerSize::Big);
-    speaker->setFileName( _problem.answer);
+    speaker->setFileName( convertToWords(TodoUtil::stoi(_problem.answer)));
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Extra space before convertToWords

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 2]
```diff
@@ -26,8 +26,39 @@
...
+                     "nine ", "ten ", "eleven ", "twelve ",
+                     "thirteen ", "fourteen ", "fifteen ",
+                     "sixteen ", "seventeen ", "eighteen ",
+                     "nineteen "
+    };
+    string ten[] = { "", "", "twenty ", "thirty ", "forty ",
+                     "fifty ", "sixty ", "seventy ", "eighty ",
+                     "ninety "
+    };
+    string numToWords(int n, string s)//Function used by convert to words
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Use better var names

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 3]
```diff
@@ -26,8 +26,39 @@
 BEGIN_NS_EGGQUIZ;
 
 static int _tryCountFishTest;
-
-Scene* EggQuizScene::createScene(string course, string level)
+ // Function and declaration for number to words conversion
+    string one[] = { "", "one ", "two ", "three ", "four ",
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Use a different name for this array

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 4]
```diff
@@ -26,8 +26,39 @@
...
+                     "ninety "
+    };
+    string numToWords(int n, string s)//Function used by convert to words
+    {
+        string str = "";
+        if (n > 19)
+            str += ten[n / 10] + one[n % 10];
+        else
+            str += one[n];
+        if (n)
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Braces for if-else

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 5]
```diff
@@ -26,8 +26,39 @@
...
+            str += s;
+        return str;
+    }
+    string convertToWords(int n) //Function for convertToWords
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Better var name

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 6]
```diff
@@ -26,8 +26,39 @@
...
+    {
+        string word;
+        word += numToWords(((n / 100) % 10), "hundred ");
+        if (n > 100 && n % 100)
+            word += "and ";
+        word += numToWords((n % 100), "");
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp]: Braces for if-else

**Review by ankitk20 — CHANGES_REQUESTED** [review]

**Review by ankitk20 — CHANGES_REQUESTED** [review]
Resolve comments

File: mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp [hunk 7]
```diff
@@ -27,35 +27,36 @@ BEGIN_NS_EGGQUIZ;
...
                      "nineteen "
     };
-    string ten[] = { "", "", "twenty ", "thirty ", "forty ",
+    string largerNumberWords[] = { "", "", "twenty ", "thirty ", "forty ",
                      "fifty ", "sixty ", "seventy ", "eighty ",
                      "ninety "
     };
-    string numToWords(int n, string s)//Function used by convert to words
+    string numToWords(int number, string s)//Function used by convert to words
     {
```
Comment (ankitk20) [inline mainapp/Classes/Games/EggQuiz/EggQuizScene.cpp:43]: Use a better name for s. Please do not use single letter names anywhere

**Review by ankitk20 — CHANGES_REQUESTED** [review]

**Review by ankitk20 — DISMISSED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**